openshift/e2e-common.sh

```diff
@@ -80,11 +80,18 @@ function timeout_non_zero() {
 function install_serverless(){
   header "Installing Serverless Operator"
 
+  GO111MODULE=off go get -u github.com/openshift-knative/hack/cmd/sobranch
+
+  local release
+  release=$(yq r "${SCRIPT_DIR}/project.yaml" project.tag)
+  release=${release/knative-/}
+  so_branch=$( $(go env GOPATH)/bin/sobranch --upstream-version "${release}")
+
   KNATIVE_EVENTING_MANIFESTS_DIR="$(pwd)/openshift/release/artifacts"
   export KNATIVE_EVENTING_MANIFESTS_DIR
 
   local operator_dir=/tmp/serverless-operator
-  git clone --branch main https://github.com/openshift-knative/serverless-operator.git $operator_dir
+  git clone --branch "${so_branch}" https://github.com/openshift-knative/serverless-operator.git $operator_dir || git clone --branch main https://github.com/openshift-knative/serverless-operator.git $operator_dir
```

Comment:
just one nit, as we don't have debug logs: can we print the checked out branch name?

Reply:
There is debug logs
```
+ KNATIVE_EVENTING_MANIFESTS_DIR=/go/src/github.com/openshift-knative/eventing/openshift/release/artifacts
+ export KNATIVE_EVENTING_MANIFESTS_DIR
+ local operator_dir=/tmp/serverless-operator
+ git clone --branch release-1.32 https://github.com/openshift-knative/serverless-operator.git /tmp/serverless-operator
Cloning into '/tmp/serverless-operator'...
fatal: Remote branch release-1.32 not found in upstream origin
+ git clone --branch main https://github.com/openshift-knative/serverless-operator.git /tmp/serverless-operator
Cloning into '/tmp/serverless-operator'...
```

Reply:
From https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift-knative_eventing/502/pull-ci-openshift-knative-eventing-release-v1.11-414-test-reconciler-aws-ocp-414/1749840014566494208/artifacts/test-reconciler-aws-ocp-414/test/build-log.txt

Reply:
Ah ok. My fault. I just checked on the build-logs from the "main" page (https://storage.googleapis.com/test-platform-results/pr-logs/pull/openshift-knative_eventing/502/pull-ci-openshift-knative-eventing-release-v1.11-411-test-reconciler-aws-ocp-411/1749840014184812544/build-log.txt) :facepalm: